Components/Adyen/PaymentMethod/PaymentMethodsProvider.php

```diff
@@ -31,16 +36,18 @@ public function __construct(
     }
 
     public function __invoke(Shop $shop): array {
-        $adyenClient = $this->adyenApiFactory->provide($shop);
-        $checkout = new Checkout($adyenClient);
-
         try {
+            $adyenClient = $this->adyenApiFactory->provide($shop);
+            $checkout = new Checkout($adyenClient);
+
             $paymentMethods = $checkout->paymentMethods([
                 'merchantAccount' => $this->configuration->getMerchantAccount($shop),
             ]);
         } catch (AdyenException $e) {
-            // TODO fix exception handling
-            return [];
+            $this->logger->error($e->getMessage(), [
+                'trace' => $e->getTraceAsString(),
+                'previous' => $e->getPrevious()
+            ]);
```

Comment:
Improve logging:

 $e->getPrevious() -> will be an object which will be quite big in the logging, 
normally here it is not required to log.

```suggestion
            $this->logger->error($e->getMessage(), [
                'merchantAccount' => $merchantAccount,
                'Shop' => $shop->getName() // validate if it is name or label or something to identify the shop instance
                'trace' => $e->getTraceAsString(),
            ]);
```